x-pack/plugins/stack_alerts/public/alert_types/geo_containment/query_builder/util_components/geo_index_pattern_select.test.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import { GeoIndexPatternSelect } from './geo_index_pattern_select';
+import { IndexPatternsContract } from 'src/plugins/data/public';
+
+class MockIndexPatternSelectComponent extends React.Component {
+  render() {
+    return 'MockIndexPatternSelectComponent';
+  }
+}
+
+function makeMockIndexPattern(id: string, fields: unknown) {
+  return {
+    id,
+    fields,
+  };
+}
+
+const mockIndexPatternService: IndexPatternsContract = ({
+  get(id: string) {
+    if (id === 'foobar_with_geopoint') {
+      return makeMockIndexPattern(id, [{ type: 'geo_point' }]);
+    } else if (id === 'foobar_without_geopoint') {
+      return makeMockIndexPattern(id, [{ type: 'string' }]);
+    }
+  },
+} as unknown) as IndexPatternsContract;
+
+test('should render without error after ', async () => {
+  const component = shallow(
+    // @ts-expect-error
+    <GeoIndexPatternSelect
+      onChange={() => {}}
+      value={'foobar_with_geopoint'}
+      includedGeoTypes={['geo_point']}
+      indexPatternService={mockIndexPatternService}
+      IndexPatternSelectComponent={MockIndexPatternSelectComponent}
+    />
+  );
+
+  // componentDidMount has async check
+  return new Promise((resolve) => {
```

Comment:
Nit: instead of this pattern, most of your async tests using the pattern

```
// Ensure all promises resolve
await new Promise((resolve) => process.nextTick(resolve));
// Ensure the state changes are reflected
component.update();
```